backend/src/main/java/ch/puzzle/pcts/service/persistence/CertificatePersistenceService.java

```diff
@@ -15,7 +23,21 @@ public CertificatePersistenceService(CertificateRepository repository) {
         this.repository = repository;
     }
 
-    public Optional<Certificate> findLeadershipExperience(Long id) {
-        return repository.findByIdAndCertificateType_CertificateKindNot(id, CertificateKind.CERTIFICATE);
+    @Override
+    public String entityName() {
+        return CERTIFICATE;
+    }
+
+    public Certificate findLeadershipExperience(Long id) {
+        return repository
+                .findByIdAndCertificateType_CertificateKindNot(id, CertificateKind.CERTIFICATE)
+                .orElseThrow(() -> {
```

Comment:
create helper method for this in the base persistance service  (throw error if not found)